libs/header/p2p/exchange.go

```diff
@@ -235,18 +236,21 @@ func (ex *Exchange[H]) performRequest(
 		return nil, fmt.Errorf("no trusted peers")
 	}
 	var reqErr error
-	for i := 0; i < len(ex.trustedPeers); i++ {
-		//nolint:gosec // G404: Use of weak random number generator
-		idx := rand.Intn(len(ex.trustedPeers))
+	peers := ex.trustedPeers
 
+	// we need this each time before calling shuffle
+	rand.Seed(time.Now().UnixNano())
```

Comment:
According to `Seed`. godoc, this will Seed global rand var. It is also deprecated since 1.20 (I know we are not there yet, but soon).  For this case we can use local random sourcer by calling:
```
rand.New(rand.NewSource(time.Now().UnixNano())).Shuffle( ... )
```